onnxruntime/core/framework/device_stream_collection.cc

```diff
@@ -78,6 +82,13 @@ class DeviceStreamCollectionImpl {
 
   size_t NumStreams() { return num_streams_; }
 
+  Stream* CreateMemoryPatternStream() {
+    if (!memory_pattern_stream_) {
```

Comment:
This "FakeStream" is relatively cheap, could we just create it once during initialization? so you don't need to expose state-update API. And you don't need to make it non-const in the following API changes.

Reply:
But DeviceStreamCollection is initialized in Session::Run() function. We can create mem_pattern_stream once the DeviceStreamCollection is created before passing DeviceStreamCollection object to the callee functions to preserve the "const" in the function signature, but mem_pattern_stream is not always necessary even if memory pattern is enabled (for example, if the Run() function is only invoked once)

Reply:
I agree. GetOrCreate() is better be avoided.